### ISSUE TITLE
Lore does not close files

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3565]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)| @fijal reported|
|-|-|
|Trac ID|trac#3565|
|Type|defect|
|Created|2008-12-05 13:56:34Z|
|Branch|https://github.com/twisted/twisted/tree/man2lore-close-files-3565|

Patch that fixes it

Attachments:

* [lore.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/eee/eee4b1e3c1664d80d2985e14ced603c223cae7f0/a47022dacab5fdb88077cc18c6412056aa8587d6.diff) (374 bytes) - added by fijal on 2008-12-05 13:56:43Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3565 3565
type__defect defect
reporter__fijal fijal
priority__normal normal
milestone__ 
branch__branches_man2lore_close_files_3565 branches/man2lore-close-files-3565
branch_author__fijal fijal
status__closed closed
resolution__fixed fixed
component__lore lore
keywords__ 
time__1228485394000000 1228485394000000
changetime__1229199458000000 1229199458000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)|@fijal removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [25570]) Branching to 'man2lore-close-files-3565'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [25571]) Apply lore.diff

refs [#3565](https://github.com/twisted/twisted/issues/3565)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @fijal<br>@exarkun set status to `new`|
|-|-|

The change is good and right AFAICT.  But it breaks all of the tests in `twisted.lore.test.test_man2lore`.  These tests use a `StringIO` as the output file.  `StringIO.close` throws away anything that's been written to the `StringIO`.  So the tests need to be updated to keep working.  They could switch to using a real file or to a `StringIO`-like which doesn't discard everything on close.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

fijal suggested using flush instead of close as a way to avoid the bad interaction between the change and the tests use of StringIO.  Build results are [here](http://buildbot.twistedmatrix.com/boxes-all?branch=/branches/man2lore-close-files-3565).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [25663]) Merge man2lore-close-files-3565

Author: fijal
Reviewer: exarkun
Fixes: [#3565](https://github.com/twisted/twisted/issues/3565)

After generating a lore document from a man page, flush the output file
to ensure the timely availability of the results.  This fixes several
lore tests on Python runtimes which don't use reference counting.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
